mitielib/src/mitie.cpp

```diff
@@ -663,7 +676,78 @@ extern "C"
             return 1;
         }
     }
-
+    
+    mitie_text_categorizer* mitie_load_text_categorizer (
+         const char* filename
+     )
+     {
+         assert(filename != NULL);
+
+         text_categorizer* impl = 0;
+         try
+         {
+             string classname;
+             impl = allocate<text_categorizer>();
+             dlib::deserialize(filename) >> classname;
+             if (classname != "mitie::text_categorizer")
+                 throw dlib::error("This file does not contain a mitie::text_categorizer. Contained: " + classname);
+             dlib::deserialize(filename) >> classname >> *impl;
+             return (mitie_text_categorizer*)impl;
+         }
+         catch(std::exception& e)
+         {
+ #ifndef NDEBUG
+             cerr << "Error loading MITIE model file: " << filename << "\n" << e.what() << endl;
+ #endif
+             mitie_free(impl);
+             return NULL;
+         }
+         catch(...)
+         {
+             mitie_free(impl);
+             return NULL;
+         }
+     }
+     
+     int mitie_categorize_text (
+         const mitie_text_categorizer* tcat_,
+         const char** tokens,
+         char** text_tag,
+         double* text_score
+     )
+     {
+         try
+         {       
+             assert(text_tag);
+             assert(text_score);
+
+             string tag;
+             double score;
+             std::vector<std::string> words;
+
+             while(*tokens)
+                 words.push_back(*tokens++);
+              
+             checked_cast<text_categorizer>(tcat_).predict(words,tag,score);
+
+             char * writable = new char[tag.size() + 1];
```

Comment:
You can't call new.  All memory needs to be allocated like other mitie objects with mitie's allocation functions.  Then it needs to be freed with those functions as well.  If this isn't done then it will lead to program crashes, especially on windows where different DLLs have separate memory allocators.